Hooked up the IdentifyObject event and returned a debug string.

OptimShi opened the pull request:
Also added the flags for when we get to the point we are passing actual data.

LtRipley36706 commented on changelog.md:
```diff
@@ -2,6 +2,12 @@
+* Added IdentifyResponseFlags, for when we start to send actual data.
+* Added a basic debug string to any ID. It simply returns the objects GUID.
+
+### 2017-04-26
```
combine

LtRipley36706 replied:
Wait, sorry, the date needs to be changed not combined. My bad.

LtRipley36706 commented on Source/ACE/Network/GameEvent/Events/GameEventIdentifyObjectResponse.cs:
```diff
@@ -1,15 +1,24 @@
+            Writer.WriteString16L("This item has a guid of " + objectID.ToString() + " (0x" + objectID.ToString("X") + ")");
         }
     }
 }
```
end on new line please

OptimShi replied:
Is there a way to add this to the StyleCop settings? It didn't throw up any alert on this. I have made a mental note, however, for the future.

LtRipley36706 commented on Source/ACE/Network/Enum/IdentifyResponseFlags.cs:
```diff
@@ -0,0 +1,25 @@
+        ResistEnchantmentBitfield = 0x400,
+        ArmorLevels = 0x4000,
+    }
+}
```
new line here as well

OptimShi commented:
Per discussion, I modified this to only spit out debug info on "DebugObjects" (as opposed to proper WorldObjects, which as of now are players, life stones and portals.) Also added the WeenieClassId for further debugging fun (which required modifying how this works a little bit, but all for the best)

Mogwai-AC commented on Source/ACE/Entity/Landblock.cs:
```diff
@@ -477,6 +477,13 @@ public void UseTime()
                     QueuedGameAction action = player.ActionQueuePop();
                     if (action != null)
                         HandleGameAction(action, player);
+                    else
```
Is there a reason we'd only process the Examination Queue when there wasn't a pending action?  I don't get the point of putting this in an else block.

OptimShi replied:
It was a way of giving lower priority to the examine request and also ensure only one action is performed per "tick" so as not to try and do too much at once. Can just as easily modify it to try and perform both an GameAction and an ExamineAction, if any, though.

Mogwai-AC commented on Source/ACE/Entity/Landblock.cs:
```diff
@@ -596,6 +603,18 @@ private void HandleGameAction(QueuedGameAction action, Player player)
                     }
+                case GameActionType.IdentifyObject:
+                    {
+                        var g = new ObjectGuid(action.ObjectId);
```
we know the retail servers throttled this, we might want to put a "TODO" comment here to that effect.
